### DISCUSSION TITLE
How can i modify filetypes for stylelint_lsp ?

### DISCUSSION THREAD

**Discussion by eulke-meli** [LazyVim/LazyVim#475] [Q&A] [answered]
Im new to lua and neovim config. I tried to config some LSPs (in that case, stylelint_lsp) without success. Currently i have this config

```lua
return {
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        rust_analyzer = {
          settings = {
            ["rust-analyzer"] = {
              lens = {
                enable = true,
              },
              checkOnSave = {
                command = "clippy",
              }
            },
          }
        },
        gopls = {
          hints = {
            assignVariableTypes = true,
            compositeLiteralFields = true,
            constantValues = true,
            functionTypeParameters = true,
            parameterNames = true,
            rangeVariableTypes = true,
          },
        },
        jsonls = {},
        html = {},
        stylelint_lsp = {
          filetypes = { "html", "css", "less", "scss" },
          settings = {
            stylelintplus = {
              autoFixOnformat = true,
              autoFixOnSave = true,
            },
          }
        },
        eslint = {},
      },
    },
    { import = "lazyvim.plugins.extras.lang.typescript" },
  }
}
```
But cannot make things work. I tried to follow the docs or view the [code](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/init.lua) without success. LSP give me completions on that places
![Captura de pantalla 2023-03-22 a la(s) 19 02 04](https://user-images.githubusercontent.com/70218774/227048586-481ffa79-02bf-470a-9579-cde044d14d01.png)

**Comment by folke** [comment, marked as answer]
What you're doing is correct. for more info, check the lspconfig docs

**Reply by eulke-meli** [reply]
I checked the lspconfig [docs for stylint_lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#stylelint_lsp) and i tried without success

**Reply by mapringg** [reply]
You might need to disable neoconf.

**Reply by folke** [reply]
@mapringg neoconf doesn't have anything to do with this afaik.

@eulke-meli what exactly isn't working? What do you expect that lsp config to do? The only thing it will do is attach to the filetypes you added. So if you do `:LspInfo` you will see that stylelint_lsp will have attached

Those `autofixonformat` and `autofixonsave` won't automatically work I think. Those are vscode settings

**Reply by eulke-meli** [reply]
I tried to dettached `stylelint-lsp` for `js` or `ts` files. I solved it making the config similar to [this](https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/extras/lang/nodejs.lua) one.
Adding this works fine

```
  -- setup lsp
  {
    "neovim/nvim-lspconfig",
    opts = {
      setup = {
        stylelint_lsp = function(_, opts)
          opts.filetypes = { "css", "scss", "less", "sass" }
        end,
      },
    }
  },

```

Thanks for the help! It was driving me crazy.
Sorry for the trouble

**Reply by danielo515** [reply]
The problem were that you were just returning a table, that was being merged instead of overwriting the config, right?
